src/components/modals/community/UpsertChatModal.tsx

```diff
@@ -0,0 +1,213 @@
+
+  const router = useRouter()
+  const { chat, hubId, onSuccess, onAfterRedirect, ...otherProps } =
+    props as UpsertChatModalProps &
+      Partial<InsertAdditionalProps & UpdateAdditionalProps>
+
+  const [createdPostId, setCreatedPostId] = useState('')
+
+  const defaultValues = {
+    image: chat?.content?.image ?? '',
+    body: chat?.content?.body ?? '',
+    title: chat?.content?.title ?? '',
+  }
+  const {
+    register,
+    control,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+    reset,
+    watch,
+  } = useForm<FormSchema>({
+    mode: 'onChange',
+    resolver: zodResolver(formSchema),
+    defaultValues,
+  })
+
+  useEffect(() => {
+    if (props.isOpen) {
+      reset(defaultValues)
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.isOpen, reset])
+
+  const texts = {
+    update: {
+      title: '✏️ Edit chat',
+      button: 'Save changes',
+    },
+    insert: {
+      title: '💬 New Chat',
+      button: 'Create',
+    },
+  }
+  const isUpdating = !!chat
+  const usedTexts = isUpdating ? texts.update : texts.insert
+
+  return (
+    <Modal
+      {...otherProps}
+      isOpen={otherProps.isOpen && !createdPostId}
+      title={usedTexts.title}
+      withCloseButton={!otherProps.onBackClick}
+    >
+      <JoinChatWrapper>
+        {({ mutateAsync }) => (
+          <UpsertPostWrapper
+            config={{
+              txCallbacks: {
+                onSuccess: async (_data, _, txResult) => {
+                  if (isUpdating) return
+
+                  const { getNewIdsFromEvent } = await import(
+                    '@subsocial/api/utils'
+                  )
+                  const [newId] = getNewIdsFromEvent(txResult)
+                  const newIdString = newId.toString()
+
+                  setCreatedPostId(newIdString)
+                  mutateAsync({ chatId: newIdString })
+
+                  setIsRedirecting(true)
+                  await router.push(
+                    urlJoin(
+                      getChatPageLink({ query: {} }, newIdString, hubId),
+                      '?new=true'
+                    )
+                  )
+                  setIsRedirecting(false)
+
+                  onAfterRedirect?.()
+                },
+              },
+            }}
+            loadingUntilTxSuccess
+          >
+            {({ isLoading: isMutating, mutateAsync }) => {
+              const onSubmit: SubmitHandler<FormSchema> = async (data) => {
+                if (isUpdating) sendEvent('click edit_chat_button')
+                else sendEvent('click create_chat_button')
+
+                await mutateAsync({
+                  spaceId: hubId,
+                  postId: chat?.id,
+                  ...data,
+                })
+                onSuccess?.()
+                if (isUpdating) props.closeModal()
+              }
+
+              const isLoading = isMutating || isRedirecting
+
+              return (
+                <form
```

Comment:
I think we need to extract form to the separate component

Reply:
done